lib/src/database/db.dart

```diff
@@ -209,7 +210,7 @@ class Db {
       }
 
       if (connection == null) {
-        connection = _masterConnection;
+        connection = _masterConnectionVerified;
```

Comment:
I think I'd wrap the null check as well. And maybe call it `isMasterConnectionValid` or something like that.